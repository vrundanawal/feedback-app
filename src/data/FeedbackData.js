const FeedbackData = [
  {
    id: 1,
    rating: 10,
    text: "Using context Api Item 1.",
  },
  {
    id: 2,
    rating: 9,
    text: "Using context Api Item 2",
  },
  {
    id: 3,
    rating: 8,
    text: "Using context Api Item 3",
  },
  {
    id: 4,
    rating: 9,
    text: "Using context Api Item 3",
  },
];

export default FeedbackData;
