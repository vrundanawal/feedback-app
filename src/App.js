import { v4 as uuidv4 } from "uuid";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { useState } from "react";
import FeedbackForm from "./components/FeedbackForm";
import FeedbackList from "./components/FeedbackList";
import FeedbackStats from "./components/FeedbackStats";
import Header from "./components/Header";

import FeedbackData from "./data/FeedbackData";
import AboutPage from "./pages/AboutPage";
import AboutIconLink from "./components/AboutIconLink";
import { FeedbackProvider } from "./context/FeedbackContext";

const App = () => {
  const [feedback, setFeedBack] = useState(FeedbackData);

  const addFeedback = (newFeedback) => {
    newFeedback.id = uuidv4();
    console.log(newFeedback);
    setFeedBack([newFeedback, ...feedback]);
  };

  const deleteFeedback = (id) => {
    //console.log("app " + id);
    if (window.confirm("Are you sure you want to delete?")) {
      setFeedBack(feedback.filter((item) => item.id !== id));
    }
  };

  return (
    <FeedbackProvider>
      <Router>
        <Header />
        <div className="container">
          <Routes>
            <Route
              exact
              path="/"
              element={
                <>
                  <FeedbackForm handleAdd={addFeedback} />
                  <FeedbackStats />
                  <FeedbackList handleDelete={deleteFeedback} />
                </>
              }
            ></Route>

            <Route path="/about" element={<AboutPage />} />
          </Routes>
          <AboutIconLink />
        </div>
      </Router>
    </FeedbackProvider>
  );
};

export default App;
