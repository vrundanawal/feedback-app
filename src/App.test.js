import { render } from "@testing-library/react";
import App from "./App";
import FeedbackData from "./data/FeedbackData";

// describe - group of related test specs => TEST SUITE

describe("AppComponent", () => {
  it("should render app comp", () => {
    expect(App).toBeTruthy();
  });

  it("should take a snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});

//test case for randomly generate uuid
jest.mock("uuid", () => {
  return {
    v4: jest.fn(() => 1),
  };
});

// test("<FeedbackList/>", () => {
//   const handleDelete = jest.fn();
//   const item = {
//     id: 12,
//     rating: 5,
//     text: "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
//   };
//   const { getByTestId } = render(
//     <FeedbackItem item={item} handleDelete={handleDelete} />
//   );
//   const deleteBtn = getByTestId("delete-list");
//   fireEvent.click(deleteBtn);
//   // expect(handleDelete).toHaveBeenCalledTimes(1);
//   expect(handleDelete).toBeCalledWith(item.id);
// });

test("Add the feedback to list", () => {
  const addFeedBack = jest.fn();
  const feedback = [];
  const newFeedBack = {
    id: 12,
    rating: 5,
    text: "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
  };

  render(<App addFeedback={[newFeedBack, ...feedback]} />);
  expect(addFeedBack).toHaveBeenCalledTimes(1);
});
