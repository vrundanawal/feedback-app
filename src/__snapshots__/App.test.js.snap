// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should take a snapshot 1`] = `
<DocumentFragment>
  <header
    style="background-color: rgba(0, 0, 0, 0.4); color: rgb(255, 106, 149);"
  >
    <div
      class="container"
    >
      <h2>
        Feedback UI
      </h2>
    </div>
  </header>
  <div
    class="container"
  >
    <div
      class="card"
      style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
    >
      <form>
        <h2>
          How would you rate your service with us?
        </h2>
        <ul
          class="rating"
        >
          <li>
            <input
              id="num1"
              name="rating"
              type="radio"
              value="1"
            />
            <label
              for="num1"
            >
              1
            </label>
          </li>
          <li>
            <input
              id="num2"
              name="rating"
              type="radio"
              value="2"
            />
            <label
              for="num2"
            >
              2
            </label>
          </li>
          <li>
            <input
              id="num3"
              name="rating"
              type="radio"
              value="3"
            />
            <label
              for="num3"
            >
              3
            </label>
          </li>
          <li>
            <input
              id="num4"
              name="rating"
              type="radio"
              value="4"
            />
            <label
              for="num4"
            >
              4
            </label>
          </li>
          <li>
            <input
              id="num5"
              name="rating"
              type="radio"
              value="5"
            />
            <label
              for="num5"
            >
              5
            </label>
          </li>
          <li>
            <input
              id="num6"
              name="rating"
              type="radio"
              value="6"
            />
            <label
              for="num6"
            >
              6
            </label>
          </li>
          <li>
            <input
              id="num7"
              name="rating"
              type="radio"
              value="7"
            />
            <label
              for="num7"
            >
              7
            </label>
          </li>
          <li>
            <input
              id="num8"
              name="rating"
              type="radio"
              value="8"
            />
            <label
              for="num8"
            >
              8
            </label>
          </li>
          <li>
            <input
              id="num9"
              name="rating"
              type="radio"
              value="9"
            />
            <label
              for="num9"
            >
              9
            </label>
          </li>
          <li>
            <input
              checked=""
              id="num10"
              name="rating"
              type="radio"
              value="10"
            />
            <label
              for="num10"
            >
              10
            </label>
          </li>
        </ul>
        <div
          class="input-group"
        >
          <input
            data-testid="text-input"
            placeholder="Write a review"
            type="text"
            value=""
          />
          <button
            class="btn btn-primary"
            disabled=""
            type="submit"
          >
            Send
          </button>
        </div>
      </form>
    </div>
    <div
      class="feedback-stats"
    >
      <h4>
        3 Reviews
      </h4>
      <h4>
        Average Rating : 9
      </h4>
    </div>
    <div
      class="feedback-list"
    >
      <div
        style="opacity: 0;"
      >
        <div
          class="card"
          style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
        >
          <div
            class="num-display"
          >
            10
          </div>
          <button
            class="close"
          >
            <svg
              color="purple"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: purple;"
              viewBox="0 0 352 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <div
            class="text-display"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. consequuntur vel vitae commodi alias voluptatem est voluptatum ipsa quae.
          </div>
        </div>
      </div>
      <div
        style="opacity: 0;"
      >
        <div
          class="card"
          style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
        >
          <div
            class="num-display"
          >
            9
          </div>
          <button
            class="close"
          >
            <svg
              color="purple"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: purple;"
              viewBox="0 0 352 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <div
            class="text-display"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. consequuntur vel vitae commodi alias voluptatem est voluptatum ipsa quae.
          </div>
        </div>
      </div>
      <div
        style="opacity: 0;"
      >
        <div
          class="card"
          style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
        >
          <div
            class="num-display"
          >
            8
          </div>
          <button
            class="close"
          >
            <svg
              color="purple"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: purple;"
              viewBox="0 0 352 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <div
            class="text-display"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. consequuntur vel vitae commodi alias voluptatem est voluptatum ipsa quae.
          </div>
        </div>
      </div>
    </div>
    <div
      class="about-link"
    >
      <a
        href="/about"
      >
        <svg
          fill="currentColor"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 384 512"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
